### ISSUE TITLE
[FR] Add hubspot webform integration

### ISSUE THREAD

**Issue by leevigraham** [verbb/formie#323]
**What are you trying to do?**

It would be great if formie could integrate with hubspot web forms. This would allow a site designer to continue to use custom form styling and still forward submissions to hubspot for additional actions.

**Comment by kylecotter** [comment]
This would be super handy for me too!

**Comment by engram-design** [comment]
Added in [1.3.18](https://github.com/verbb/formie/releases/tag/1.3.18)

**Comment by kylecotter** [comment]
@engram-design Thanks for this! Just tested it out. I have a Formie form with just an email field. And a Hubspot form with just an email field. Setting the Formie form with "Map to Form" enabled and the email fields mapped to each other results in a "400 Bad Request The request is not valid" as the response from the queue job. Unfortunately any other meaningful error message seems to have been truncated in the response.

Also tried in combo with "Map to Contact" enabled to the same result.

Using "Map to Contact" on its own works as it did previously.

This is with Formie 1.3.18 and Craft 3.6.6

Any ideas? Thanks!

**Comment by engram-design** [comment]
@kylecotter Are you able to have a look at the log files (namely formie.log and queue.log) and see if there's anything in there?

**Comment by kylecotter** [comment]
@engram-design yep! I think these are the relevant bits.

formie.log

```
2021-02-23 00:55:01 [-][-][-][error][formie] Hubspot: API error: “Client error: `POST https://api.hsforms.com/submissions/v3/integration/submit/432152/cc3e484d-36a7-4deb-94be-deb71a840012` resulted in a `400 Bad Request` response:
{"status":"error","message":"The request is not valid","correlationId":"53686059-271a-467c-8c92-2cc0cb6e4e04","errors":[ (truncated...)
” /var/www/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
```

queue log

```
2021-02-23 01:00:43 [-][-][-][error][verbb\formie\errors\IntegrationException] verbb\formie\errors\IntegrationException: API error: “Client error: `POST https://api.hsforms.com/submissions/v3/integration/submit/432152/cc3e484d-36a7-4deb-94be-deb71a840012` resulted in a `400 Bad Request` response:
{"status":"error","message":"The request is not valid","correlationId":"cba665ee-8c83-40e7-a918-8b878ef4bdc8","errors":[ (truncated...)
” /var/www/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113 in /var/www/vendor/verbb/formie/src/base/Integration.php:137
Stack trace:
#0 /var/www/vendor/verbb/formie/src/integrations/crm/HubSpot.php(336): verbb\formie\base\Integration::error(Object(verbb\formie\integrations\crm\HubSpot), 'API error: \xE2\x80\x9CC...', true)
#1 /var/www/vendor/verbb/formie/src/services/Submissions.php(211): verbb\formie\integrations\crm\HubSpot->sendPayload(Object(verbb\formie\elements\Submission))
#2 /var/www/vendor/verbb/formie/src/jobs/TriggerIntegration.php(44): verbb\formie\services\Submissions->sendIntegrationPayload(Object(verbb\formie\integrations\crm\HubSpot), Object(verbb\formie\elements\Submission))
#3 /var/www/vendor/yiisoft/yii2-queue/src/Queue.php(246): verbb\formie\jobs\TriggerIntegration->execute(Object(craft\queue\Queue))
#4 /var/www/vendor/yiisoft/yii2-queue/src/cli/Queue.php(162): yii\queue\Queue->handleMessage('61', 'O:36:"verbb\\for...', '300', '1')
#5 /var/www/vendor/yiisoft/yii2-queue/src/cli/Command.php(146): yii\queue\cli\Queue->execute('61', 'O:36:"verbb\\for...', '300', '1', NULL)
#6 [internal function]: yii\queue\cli\Command->actionExec('61', '300', '1', '0')
```

Thanks!

**Comment by engram-design** [comment]
Hmmm, yep looks like the response is truncated, I'll see what I can do about that.

**Comment by engram-design** [comment]
In the meantime, can you confirm if there are any required fields in your HubSpot form you're not mapping to?

**Comment by leevigraham** [comment]
@kylecotter  Also check that the email you used is a valid email… pretty sure hubspot checks the domain is real and it doesn't like aliases.

**Comment by kylecotter** [comment]
@engram-design I just pulled your latest commit in and was able to output an actual error `{"message":"Error in 'context.ipAddress'. IP Address '' is invalid","errorType":"INVALID_IP_ADDRESS"}`, so thanks for that update!

I'm seeing if there's something on my side not configured right. Was trying to keep my testing simple so it's literally just an email field on both the Formie side and the Hubspot side mapping to each other.

<img width="800" alt="Screen Shot 2021-02-23 at 3 38 25 PM" src="https://user-images.githubusercontent.com/505894/108905192-6957e900-75ed-11eb-80a4-0985895691db.png">

@leevigraham thanks for the input! Confirmed still happening with a real email and Hubspot whitelist settings are appropriate.
